### ISSUE TITLE
An estimation for a planet/star system's spin-orbit resonance regime

### ISSUE THREAD

**Issue by nichollsh** [FormingWorlds/PROTEUS#63]
Currently, we provide the orbital distance and stellar mass as input parameters. This provides the length of the planet's year. We also input the zenith angle and heat redistribution factor, which allow us to handle resonant and asynchronously rotating cases.

Determining which planets are spin-orbit resonant and which are not is far from trivial, so in some cases it isn't clear which input parameters should be chosen. It would be helpful to implement a simple model for determining which state a given configuration is *likely* to enter into. This will fall short of a robust model which actually estimates the tidal stress on the planet and thereby more realistically determines the rotational regime.

This could be included in the main code itself, or simply as a tool which provides information to the user before the model is executed.

**Comment by nichollsh** [comment]
One simple way to approach this is to calculate the locking time-scale. This would require some estimations (guesses) for the properties of the planet (e.g. of its rigidity and ability to dissipate energy imparted by the tidal stresses).

https://en.wikipedia.org/wiki/Tidal_locking#Timescale

**Comment by nichollsh** [comment]
Recent paper related to this issue: https://arxiv.org/abs/2410.05373

**Comment by nichollsh** [comment]
https://arxiv.org/pdf/2507.20093

**Comment by nichollsh** [comment]
I believe that this issue is closely related to the formulation that we discussed with @mohammadfarhat1 last week, @MarijnJ0 @timlichtenberg. What do you think? Should we adapt this issue or close it, to favour a different one about incorporating the tidal-orbital-spin calculation (described by Mohammad) into Obliqua or PROTEUS?

**Comment by timlichtenberg** [comment]
It is closely related but I am not sure we need to close it. We will probably need a bunch of descriptions for the orbital evolution, not just one. The one we are currently aiming to implement may make substantial simplifications that are suitable for the Earth-Moon system, but not for multi-body exoplanet systems.

**Comment by MarijnJ0** [comment]
If I understand correctly, we need to update the zenith angle and heat distribution factor on-the-fly. So when we couple PROTEUS to the orbital dynamics model we need some set of equations that can deternine these quantities. Then, by extension, it would also be useful to implement a script that can determine the (time averaged / final?) orbital state of the system a priori. This would allow the user to determine an appropriate zenith angle and heat distribution factor without the need to model the orbital dynamics. In this case, I presume these quantities remain fixed throughtout the entire simulation. 

Given this I agree with Tim, this would definitely still be useful for causes where modeling orbit+tides is unnecessary. 

@nichollsh do you know if there is currently a way to infer the zenith angle and heat distribution factor from the orbital parameters in PROTEUS?

**Comment by nichollsh** [comment]
Yes, that's exactly the idea. Although I should note that the scale factor I am referring to is not a heat redistribution factor, it's a geometric scale factor which represents the spherical geometry of the radiation field. We need to have some flexibility, too, so I suppose the combinations are:

a) Orbit/spin evolution disabled
  1) ZA and SF set by user in config
  2) ZA and SF calculated from planet's (fixed) rotation and orbital period

b) Orbit/spin evolution enabled
  1) ZA and SF set by user in config
  2) ZA and SF calculated self-consistently based on rotational evolution


Methods for handling (a2) and (b2) are what we are currently missing.

As far as I can tell, there's no literature describing how to calculate these quantities robustly. We could derive something ourselves, to obtain the 'average' value of ZA and SF given a planet's orbital and axial periods? 

I'd recommend reading this paper, if you're interested: https://journals.ametsoc.org/view/journals/atsc/71/8/jas-d-13-0392.1.xml

**Comment by MarijnJ0** [comment]
Yeah that sounds good! I think we can definitely derive something ourselves. The paper seems to also suggest that these quantities will depend on the atmosphere of the planet, do you think we should derive them based on only the orbital parameters? or also include some corrections due to the atmosphere (e.g. at the previous timestep).

**Comment by nichollsh** [comment]
Yes, this is a key aspect that we need to consider. If you were to consider an asynchronously rotating planet (e.g. the Earth), a simple geometric approach for determining the average Zenith angle would leave you with $\theta=90^\circ$ (i.e. on the day/night terminator), which is not very useful. This is why we introduce the geometric scale factor of 1/4 and instead determine the average dayside zenith angle.

I believe that there is not going to be a 'correct' way to perform this angle-averaging, since we are ultimately representing a 3D geometry in 1D. Doesn't seem like something we can determine from atmospheric properties. Perhaps we could simply adopt the 1/4 geometric scale factor, and then ask "what is the average Zenith angle on the dayside only?" to derive some type of formulation.
